org.eclipse.winery.topologymodeler.ui/src/app/canvas/canvas.component.ts

```diff
@@ -0,0 +1,1403 @@
+        this.unbindNewNodeMouseMove();
+        this.unbindNewNodeMouseUp();
+        setTimeout(() => this.newJsPlumbInstance.revalidate(this.newNode.id), 1);
+        this.repaintJsPlumb();
+    }
+
+    /**
+     * Setter for PaletteState
+     * @param currentPaletteOpened
+     */
+    setPaletteState(currentPaletteOpened: boolean): void {
+        if (currentPaletteOpened) {
+            this.currentPaletteOpenedState = currentPaletteOpened;
+        }
+    }
+
+    /**
+     * Gets called if relationships get deleted or created and calls the
+     * correct handler.
+     * @param currentRelationships  List of all displayed relationships.
+     */
+    updateRelationships(currentRelationships: Array<TRelationshipTemplate>): void {
+        if (currentRelationships.length > 0) {
+            if (currentRelationships.length !== this.allRelationshipTemplates.length) {
+                const difference = currentRelationships.length - this.allRelationshipTemplates.length;
+                if (difference === 1) {
+                    this.handleNewRelationship(currentRelationships);
+                } else if (difference < 1) {
+                    this.handleDeletedRelationships(currentRelationships);
+                } else {
+                    try {
+                        this.handleLoadedRelationships(currentRelationships);
+                    } catch (e) {
+                    }
+                }
+            } else {
+                this.updateRelName(currentRelationships);
+            }
+        }
+    }
+
+    /**
+     * Handler for new relations, adds it to the internal representation
+     * @param currentRelationships  List of all displayed relations.
+     */
+    handleNewRelationship(currentRelationships: Array<TRelationshipTemplate>): void {
+        const newRel = currentRelationships[currentRelationships.length - 1];
+        this.allRelationshipTemplates.push(newRel);
+        console.log(newRel);
+        this.manageRelationships(newRel);
+    }
+
+    /**
+     * Handler for deleted relations, removes it from the internal representation
+     * @param currentRelationships  List of all displayed relations.
+     */
+    handleDeletedRelationships(currentRelationships: Array<TRelationshipTemplate>): void {
+        for (const rel of this.allRelationshipTemplates) {
+            if (!currentRelationships.map(con => con.id).includes(rel.id)) {
+                const deletedRel = rel.id;
+                const index = this.allRelationshipTemplates.map(con => con.id).indexOf(deletedRel);
+                this.allRelationshipTemplates.splice(index, 1);
+            }
+        }
+    }
+
+    /**
+     * Handles relationships internally (store)
+     * @param currentRelationships  List of all displayed relations.
+     */
+    handleLoadedRelationships(currentRelationships: Array<TRelationshipTemplate>): void {
+        try {
+            this.allRelationshipTemplates = currentRelationships;
+            setTimeout(() => {
```

Comment:
the timeout is needed in cases the nodes are loaded before the relationships, because the relationships need a source and a target node for establishing a connection